nixos.nix

```diff
@@ -124,14 +151,48 @@ in
             directories = cfg.${persistentStoragePath}.directories;
             files = unique (map dirOf cfg.${persistentStoragePath}.files);
           in
-          nameValuePair
-            "createDirsIn-${replaceStrings [ "/" "." " " ] [ "-" "" "" ] persistentStoragePath}"
-            (noDepEntry (concatMapStrings
-              (mkDirWithPerms persistentStoragePath)
-              (directories ++ files)
-            ));
+          {
+            "createDirsIn-${replaceStrings [ "/" "." " " ] [ "-" "" "" ] persistentStoragePath}" =
+              noDepEntry (concatMapStrings
+                (mkDirWithPerms persistentStoragePath)
+                (directories ++ files));
+          };
+
+        dirCreationScripts = foldl' recursiveUpdate { } (map mkDirCreationScriptForPath persistentStoragePaths);
+
+        bindOrLinkFileScript = persistentStoragePath: file:
+          let
+            targetFile = escapeShellArg (concatPaths [ persistentStoragePath file ]);
+            mountPoint = escapeShellArg file;
+          in
+          {
+            "bindOrLink-${sanitizeName targetFile}" = {
+              deps = [ "createDirsIn-${replaceStrings [ "/" "." " " ] [ "-" "" "" ] persistentStoragePath}" ];
+              text = ''
```

Comment:
I wonder if you {sh,c}ould abstract this out, since it looks like it's used here, and on line 77 (the script contents, at least). Otherwise, they risk falling out of sync and creating potentially difficult-to-debug issues.

Reply:
(In case it wasn't clear -- "this" being the script's text)

Reply:
Yeah, true! I'll deduplicate this.

Reply:
Done!